### PR TITLE
Check auth in hoc

### PR DESCRIPTION
**Pull request by tenshiemi**

### CODE REVIEW

File: app/utils/requireAuth.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    componentDidMount() {
+      this.checkAndRedirect();
+    }
+    componentDidUpdate() {
+      this.checkAndRedirect();
+    }
+    checkAndRedirect() {
+      if (!this.state.authToken) {
+        this.context.router.push('/start');
+      }
```
Comment (ashley-jelks-truss) [inline app/utils/requireAuth.js]: If the authToken check is invalid then what is the router context?

Comment (tenshiemi) [inline app/utils/requireAuth.js]: It will continue to the original route

File: app/utils/requireAuth.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    }
+    render() {
+      return (
+        <div className="authenticated">
+          {this.state.authToken ? <Component {...this.props} /> : null}
```
Comment (ashley-jelks-truss) [inline app/utils/requireAuth.js]: How is this spread operator working here?....If the authToken is valid what props are passed to <Component> and from where?

Comment (tenshiemi) [inline app/utils/requireAuth.js]: It passes through both the props from the original component and the higher order component.

**Pull request merged**